README.md

```diff
@@ -1,41 +1,143 @@
+Python script to generate UE4 .h and .cpp files for UnrealEngine to interface with ROS2 messages.
 
-Based on Jinja2.
+\* Based on Jinja2 to generatee msg
 
-Ignores message types deprecated in Foxy.
+\* Ignores message types deprecated in Foxy.
 
 ### Usage
-`python3 CodeGeneratorFromPath.py <ros2-base-share-path> [<custom-message-path>]`
 
-`ros2-base-share-path` is used to find base definition of types
-`custom-message-path` is the path of messages that need to be generated (multiple are allowed)
+#### Generate source files
+1. cd UE_tools/CodeGen
+2. python3 gen_ue_from_ros.py --module <UE module name used in class/struct definition> --dependency <path to ros2 dependency pkgs> --target <names of target pkgs>
+
+    e.g.
+
+    `python3 gen_ue_from_ros.py`
+
+    \* default dependency and target pkgs are defined at top of gen_ue_from_ros.py
+
+#### Copy source files to Unreal Project
+1. cd UE_tools/CodeGen
+2. python3 post_genrate.py --ue_proj_path <path to UnrealEngine project> 
+
+    e.g.
+
+    `python3 post_generate.py --ue_proj_path /home/user/turtlebot3-UE`
+
+
+
+### Scripts
+- **gen_ue_from_ros.py**: 
+
+    Generate .h and cpp files for UnrealEngine project from msg/srv/action definitions. Generated class/struct can be used from UnrealEngine C++ and Blueprint. 
+    
+    All properties can be accessible from C++. Not all properties but only Blueprint comaptible type member can be accesible from  Blueprint.
+
+    If your custom msg need other msg dependency, you need to specify that dependnecy from arg.
+
+    \* Default dependencies and targets are defined near the top of files.
 
-e.g. `python3 CodeGeneratorFromPath.py /opt/ros/foxy/share/ /opt/ros/foxy/share/geometry_msgs/` generates `geometry_msgs` using definitions in `/opt/ros/foxy/share/` and `/opt/ros/foxy/share/geometry_msgs/`
 
-#### For usage with rclUE
-1. `python3 post_generate.py <path to ue project whith have rclUE under Plugins> <path to install> <ros_package_name>`. e.g.
+    \* Please check [limitiations](#limitation) as well.
 
-    - installed package `python3 post_generate.py /home/yu/io_amr_UE /opt/ros/foxy example_interfaces`. *you need to install example_interfaces by sudo `apt install ros-foxy-example-interfaces`
-    - locally build package `python3 post_generate.py /home/yu/io_amr_UE /home/yu/test_ws/install/ue_msgs ue_msgs`
+- **post_generate.py**: 
 
-2. The package name needs to be added to `rclUE.Build.cs` and `ros2lib.Build.cs`, in the `folders` array.
-3. Update the `LD_LIBRARY_PATH` in `run_editor.sh` script
+    Copy generated .h and cpp to target UnrealEngine project folder such as **turtlebot3-UE/Plugin/rclUE/Source/Thirdparty/ros**. 
 
-Please check the corresponding files and folders in the rclUE Plugin to see how other autogenerated files are handled.
+    \* Black list is deifined near the beginning of the script which are not copied to Unreal project since there are [Limitiations](#limitation)
```

Comment:
```suggestion
    \* Black list is defined near the beginning of the script which are not copied to Unreal project since there are [Limitations](#limitation)
```